File: 205. Isomorphic Strings.py
class Solution(object):
    def isIsomorphic(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        dic_s = {}
        dic_t = {}
        for i,j in zip(s, t):
            if i not in dic_s and j not in dic_t:
                dic_s[i] = j
                dic_t[j] = i
            elif (i in dic_s and j not in dic_t) or (i not in dic_s and j in dic_t):
                return False
            else:
                if dic_s[i] != j or dic_t[j] != i:
                    return False
        return True

File: 205. Isomorphic Strings.cpp
#include <unordered_map>
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        unordered_map <char, char> dic_s, dic_t;
        for (int i = 0; i < s.length(); i++)
        {
            if (dic_s.count(s[i]) == 0 && dic_t.count(t[i]) == 0)
            {
                dic_s[s[i]] = t[i];
                dic_t[t[i]] = s[i];
            }
            else if (dic_s.count(s[i]) > 0 && dic_t.count(t[i]) > 0)
            {
                if (dic_s[s[i]] != t[i] || dic_t[t[i]] != s[i]) return false;
            }
            else
                return false;
        }
        return true;
    }
};
